File: course/src/main/java/com/ead/course/configs/security/AuthenticationJwtFilter.java
package com.ead.course.configs.security;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
import org.springframework.util.StringUtils;
import org.springframework.web.filter.OncePerRequestFilter;

import javax.servlet.FilterChain;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.util.UUID;


public class AuthenticationJwtFilter extends OncePerRequestFilter {

    @Autowired
    private JwtProvider jwtProvider;

    private final Logger log = LogManager.getLogger(this.getClass().getName());


    @Override
    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
        try {
            var jwtStr = getToken(request);

            if (jwtStr != null && jwtProvider.validateJwt(jwtStr)) {
                var userId = jwtProvider.getSubjectJwt(jwtStr);
                var rolesStr = jwtProvider.getClaimNameJwt(jwtStr, "roles");
                var userDetails = UserDetailsImpl.build(UUID.fromString(userId), rolesStr);
                var authentication = new UsernamePasswordAuthenticationToken(
                        userDetails, null, userDetails.getAuthorities()
                );
                authentication.setDetails(new WebAuthenticationDetailsSource().buildDetails(request));
                SecurityContextHolder.getContext().setAuthentication(authentication);
            }
        } catch (Exception e) {
            log.error("Cannot set User Authentication: {}", e);
        }

        filterChain.doFilter(request, response);
    }

    private String getToken(HttpServletRequest request) {
        String headerAuth = request.getHeader("Authorization");
        if (StringUtils.hasText(headerAuth) && headerAuth.startsWith("Bearer ")) {
            return headerAuth.substring(7, headerAuth.length());
        }
        return null;
    }
}

File: notification-hex/src/main/kotlin/com/ead/notificationhex/adapters/configs/security/AuthenticationJwtFilter.kt
package com.ead.notificationhex.adapters.configs.security

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.security.web.authentication.WebAuthenticationDetailsSource
import org.springframework.stereotype.Component
import org.springframework.util.StringUtils
import org.springframework.web.filter.OncePerRequestFilter
import java.util.*
import javax.servlet.FilterChain
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

@Component
class AuthenticationJwtFilter() : OncePerRequestFilter() {

    @Autowired
    lateinit var jwtProvider: JwtProvider

    override fun doFilterInternal(
        request: HttpServletRequest,
        response: HttpServletResponse,
        filterChain: FilterChain
    ) {
        try {
            val jwtStr = getToken(request)

            if (jwtStr != null && jwtProvider.validateJwt(jwtStr)) {
                val userId = jwtProvider.getSubjectJwt(jwtStr)
                val rolesStr = jwtProvider.getClaimNameJwt(jwtStr, "roles")
                val userDetails = UserDetailsImpl.build(UUID.fromString(userId),rolesStr)
                val authentication = UsernamePasswordAuthenticationToken(
                    userDetails, null, userDetails.authorities
                )

                authentication.details = WebAuthenticationDetailsSource().buildDetails(request)
                SecurityContextHolder.getContext().authentication = authentication
            }
        } catch (e: Exception) {
            println("Cannot set user Authentication: $e")
        }

        filterChain.doFilter(request, response)
    }

    fun getToken(request: HttpServletRequest) : String? {
        val headerAuth = request.getHeader("Authorization")

        if(StringUtils.hasText(headerAuth) && headerAuth.startsWith("Bearer ")) {
            return headerAuth.substring(7, headerAuth.length)
        }
        return null
    }
}